datumaro/components/format_detection.py

```diff
@@ -116,16 +116,17 @@ def _is_path_within_root(self, path: str) -> bool:
 
         return True
 
-    def _start_requirement(self) -> None:
+    def _start_requirement(self, req_type: str) -> None:
         assert not self._alternatives_context, \
-            "checks can't be done directly within 'alternatives' blocks"
+            f"a requirement ({req_type}) can't be placed directly within " \
+            "an 'alternatives' block"
 
     def fail(self, requirement: str) -> NoReturn:
         """
         Places a requirement that is never met. `requirement` must contain
         a human-readable description of the requirement.
         """
-        self._start_requirement()
+        self._start_requirement("fail")
```

Comment:
"a requirement ('fail') can't be placed directly within an 'alternatives' block"

Still sound strange, IMO. Not a blocker, anyway.

Reply:
🤷‍♂️ It's a message for developers. I think they can figure it out.